### DISCUSSION TITLE
How do I understand Node's module resolution in pnpm?

### DISCUSSION THREAD

**Discussion by ominus3** [pnpm/pnpm#8347] [Q&A] [answered]
I have confusion in this [symlinked-node-modules-structure](https://pnpm.io/symlinked-node-modules-structure) article, If node ignores symlinks, why require('bar') from foo don't look up in upper folder .pnpm/node_modules(which don't exist), instead resolve with ./pnpm/bar@1.0.0/node_modules/bar?
<img width="912" alt="image" src="https://github.com/user-attachments/assets/7a7eb92f-e256-4b17-9ad0-ff4b705f38cf">

**Comment by zkochan** [comment, marked as answer]
It doesn't ignore symlinks when searching for dependencies. It will resolve a dependency from a symlink. But the location of the resolved dependency won't be the symlink but the directory that the symlink points to.

**Reply by ominus3** [reply]
Thanks a lot，the doc needs update？
